### PR TITLE
Eliminates caching from our CI setup, moves to Java 11

### LINKED ISSUE #211: Github Actions caching causes at least 1 shard in most CI builds to time out
We should probably just get rid of our fancy caching strategy altogether, it's clearly extremely unreliable.

### PR DESCRIPTION
**Pull request by rjrjr**
Caused timeouts more often than not.

closes #211

Also a second commit to move to JDK 11.

### CODE REVIEW

File: .github/workflows/kotlin.yml [hunk 1]
```diff
@@ -10,58 +10,9 @@ env:
...
-          ls -lhrt $GRADLE_HOME/caches || true
-
-      ## Actual task
-      - name: Assemble with gradle
-        run: ./gradlew assemble --build-cache --no-daemon --stacktrace --gradle-user-home "$GRADLE_HOME"
-
-      # This should ideally be done in the Check job below, but until gradle caching is fixed we
-      # need to do it after assembling. See https://github.com/square/workflow/issues/1152.
-      - name: Run dokka to validate kdoc
-        run: ./gradlew dokka siteDokka --build-cache --no-daemon --stacktrace --gradle-user-home "$GRADLE_HOME"
```
Comment (zach-klippenstein) [inline .github/workflows/kotlin.yml:59]: Don't forget to add this `siteDokka` task to the list below.

File: .github/workflows/kotlin.yml [hunk 2]
```diff
@@ -78,29 +29,20 @@ jobs:
...
-        with:
-          path: ${{ env.GRADLE_HOME }}/caches
-          # Don't set restore-keys so cache is always only valid for the current build config.
-          # Also don't use ${{ runner.os }} in the key so we don't re-assemble for UI tests.
-          key: gradle-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle*') }}-${{ hashFiles('**/buildSrc/**') }}-${{ github.sha }}
-
       ## Actual task
       - name: Check with Gradle
-        run: ./gradlew ${{ matrix.gradle-task }} --build-cache --no-daemon --stacktrace --gradle-user-home "$GRADLE_HOME"
+        run: ./gradlew ${{ matrix.gradle-task }} --no-daemon --stacktrace --gradle-user-home "$GRADLE_HOME"
```
Comment (zach-klippenstein) [inline .github/workflows/kotlin.yml:42]: I think you can drop the `--gradle-user-home` args too, and get rid of that envar definition at the top of the file.

**Review by zach-klippenstein — APPROVED** [review]

File: .github/workflows/kotlin.yml (hunk 1 above)
Comment (rjrjr) [reply .github/workflows/kotlin.yml:59]: Nice catch

File: .github/workflows/kotlin.yml [hunk 3]
```diff
@@ -6,62 +6,31 @@ on:
       - main
   pull_request:
 
-env:
-  GRADLE_HOME: ${{ github.workspace }}/gradle-home
-
 jobs:
-  assemble:
+  dokka:
     name: Assemble
```
Comment (rjrjr) [inline .github/workflows/kotlin.yml:11]: s/Assemble/Dokka/

**Review by zach-klippenstein — APPROVED** [review]

**Pull request merged**